js/modules/k6/experimental/tracing/sampling.go

```diff
@@ -0,0 +1,64 @@
+package tracing
+
+import "math/rand"
+
+// Sampler is an interface defining probabilistic sampling strategies.
+type Sampler interface {
```

Comment:
I suggest you look at the `Sampler` interface definition the Open Telemetry Go SDK provides. I think it's a bit more powerful:

https://github.com/open-telemetry/opentelemetry-go/blob/3015c86cfd1e1868b9c57a998d44ebe25acffedc/sdk/trace/sampling.go#L27-L54.

Reply:
Thanks for the pointer 🙇🏻 It looks thorough indeed, but I don't necessarily see anything that would need to introduce at the moment though. Did you have anything specific in mind? 🤓

Reply:
The `Sampler` interface you proposed seems to be coupled with "probabilistic" sampling. For example, you define the `SetSamplingRate(percent float64)` method, which only makes sense for "probabilistic" sampling.

The interface that Open Telemetry has is much more flexible. With `ShouldSample` method accepting the `SamplingParameters`, it's much easier to implement a struct implementing this interface that makes decisions based on more parameters.

Reply:
I see your point then. You're right in that sense it's more flexible indeed 👍🏻 gonna give a shot at taking inspiration from it in this PR then. Thanks for your input 🙇🏻

Reply:
Adopted the same approach 👍🏻 Thanks for the pointer. I did not add a Parameters type, as we really don't need it yet at the moment, but it shouldn't be a problem to add it at a later point in time if the requirement changes 🙇🏻